program/rust/src/error.rs

```diff
@@ -7,31 +7,37 @@ use thiserror::Error;
 pub enum OracleError {
     /// Generic catch all error
     #[error("Generic")]
-    Generic                        = 600,
+    Generic                              = 600,
     /// integer casting error
     #[error("IntegerCastingError")]
-    IntegerCastingError            = 601,
+    IntegerCastingError                  = 601,
     /// c_entrypoint returned an unexpected value
     #[error("UnknownCError")]
-    UnknownCError                  = 602,
+    UnknownCError                        = 602,
     #[error("UnrecognizedInstruction")]
-    UnrecognizedInstruction        = 603,
+    UnrecognizedInstruction              = 603,
     #[error("InvalidFundingAccount")]
-    InvalidFundingAccount          = 604,
+    InvalidFundingAccount                = 604,
     #[error("InvalidSignableAccount")]
-    InvalidSignableAccount         = 605,
+    InvalidSignableAccount               = 605,
     #[error("InvalidSystemAccount")]
-    InvalidSystemAccount           = 606,
+    InvalidSystemAccount                 = 606,
     #[error("InvalidWritableAccount")]
-    InvalidWritableAccount         = 607,
+    InvalidWritableAccount               = 607,
     #[error("InvalidFreshAccount")]
-    InvalidFreshAccount            = 608,
+    InvalidFreshAccount                  = 608,
     #[error("InvalidInstructionVersion")]
-    InvalidInstructionVersion      = 609,
+    InvalidInstructionVersion            = 609,
     #[error("InstructionDataTooShort")]
-    InstructionDataTooShort        = 610,
+    InstructionDataTooShort              = 610,
     #[error("InstructionDataSliceMisaligned")]
-    InstructionDataSliceMisaligned = 611,
+    InstructionDataSliceMisaligned       = 611,
+    #[error("DeserializationError")]
+    DeserializationError                 = 612,
+    #[error("FailedAuthenticatingUpgradeAuthority")]
+    FailedAuthenticatingUpgradeAuthority = 613,
+    #[error("FailedPdaVerification")]
+    FailedPdaVerification                = 614,
```

Comment:
`InvalidUpgradeAuthority` and `InvalidPda` would be better names: "failed" refers to the consequence of the error being thrown, not the error itself.

Reply:
Yes